### ISSUE TITLE
About STAR-Prompt 

### ISSUE THREAD

**Issue by kkkcLi** [aimagelab/mammoth#60]
Thanks for your excellent works! However,I have a question about STAR-Prompt. I came across a paper at ICLR 2025 that is very similar to your work, and some of the reviewers raised concerns about the fairness of incorporating CLIP.
For instance,"The use of CLIP’s text and image encoders, which contain extensive pre-trained knowledge, raises concerns about data overlap. The testing data may overlap with or be highly correlated to CLIP’s training data, making the observed performance gains somewhat expected. It’s unclear if the improvement is due to the novel aspects of TIPS or simply the inclusion of CLIP, as other methods might similarly benefit from using CLIP. This ambiguity makes it unclear to identify the key design elements driving performance improvement." 
How do you view this issue? Did you encounter similar questions during the submission process?
I am looking forward to your reply. Thank you!

**Comment by loribonna** [comment]
Hi @kkkcLi! It seems that your question is related to [this paper currently under review](https://openreview.net/forum?id=QYgtZRTv3e). We had a similar experience in a previous submission of STAR-Prompt. At the time, we included datasets that significantly differ from the CLIP's pretraining data in the rebuttal, but we ended up rejected anyway.

In the new submission we decided to cover as much datasets as possible (medical and fine-grained datasets, as suggested by the reviewer of the paper you referenced, but also satellite and aerial datasets). Since the real CLIP pretraining data is private we added those datasets that have a low zero-shot performance.

**Comment by kkkcLi** [comment]
Thank you for your patient response! However, I still have a small question. In Star-Prompt, does the CLIP model in the first stage have to be trained before the second stage? And in the learning process of the second stage, does it always use the output of the CLIP model that has already been trained in the first stage, regardless of whether Confidence Modulation is used? I’m not sure if my understanding is correct. If it’s wrong, please kindly point out the mistake.

**Comment by loribonna** [comment]
In general, yes, we train the first stage prompts for a couple of epochs before training the second stage and we use the outputs of the trained first stage (with confidence modulation, as in Eq. 5 of the paper) to tune the second ViT.

We also evaluated without training the first stage (tab. 3 of the paper, the line "w/o first-level prompts"). We found that it works fine in some scenarios (ISIC) but on others it significantly reduces performance (Imagenet-R).

**Comment by kkkcLi** [comment]
Thank you for your prompt response! I noticed that the residual prompt used in the second-stage training does not depend on the true class of the data but rather on the prediction results of CLIP (which can be either the one trained in the first stage or not). This is my understanding—am I correct?

**Comment by loribonna** [comment]
Yes, we use the predictions of CLIP since that is what would occur during inference too. 
We use the true labels only for the CE loss.

During the development I think we also tried to use the true labels instead of CLIP but we got way worse accuracy. 
I might be wrong though, it's been several months since that.

**Comment by kkkcLi** [comment]
Thank you very much

**Comment by loribonna** [comment]
No problem! Feel free to reopen the issue if you have more questions.

**Comment by kkkcLi** [comment]
Sorry to bother you again. Still about STAR-Prompt,I’m quite curious about something—when you submitted your paper, did the reviewers raise any problems about the training time cost and testing time cost of your method?
